pyface/util/fix_introspect_bug.py

```diff
@@ -78,7 +78,11 @@ def getAttributeNames(object, includeMagic=1, includeSingle=1,
     # e.g. ITK http://www.itk.org/
     attributes = [attribute for attribute in attributes \
                   if isinstance(attribute, six.string_types)]
-    attributes.sort(lambda x, y: cmp(x.upper(), y.upper()))
+    if six.PY3:
```

Comment:
```suggestion
    attributes.sort(key=str.upper)
```
I think that this can be done much more simply using `str.upper` on all supported versions of Python.

Reply:
`key=str.upper` won't work for sorting Unicode strings on Python 2.

Reply:
True.  I guess a `key=lambda x: s.upper()` is the simplest way forward then.